src/scripting/game_lua_kernel.cpp

```diff
@@ -4260,6 +4248,13 @@ game_lua_kernel::game_lua_kernel(game_state & gs, play_controller & pc, reports
 		{"zoom", &dispatch<&game_lua_kernel::intf_zoom>},
 		{"clear_menu_item", &dispatch<&game_lua_kernel::intf_clear_menu_item>},
 		{"set_menu_item", &dispatch<&game_lua_kernel::intf_set_menu_item>},
+		{"allow_end_turn", &dispatch<&game_lua_kernel::intf_allow_end_turn>},
+		{"allow_undo", &dispatch<&game_lua_kernel::intf_allow_undo>},
+		{"cancel_action", &dispatch<&game_lua_kernel::intf_cancel_action>},
```

Comment:
I'm hesitant about classing `allow_undo` and `cancel_action` as "functions for querying or manipulating the in-game UI".

* `cancel_action` seems more like a unit-placement function, putting the unit in the hex that it's currently moving through.
* `allow_undo`'s effect on the event stack will affect the UI, but it feels like a property of the `[event]` saying that the `[event]` doesn't affect game state.

Reply:
Hmm… I definitely see your point. Perhaps allow_undo's deprecation should instead be integrated into the events PR using a new API.

I don't see `cancel_action` as a unit function. It also feels more event-related, actually, so maybe it should also be moved into game_events as part of the events PR?

Reply:
Classing them as event-related sounds good to me.

Redesigning this bit of the API sounds good too.

Reply:
If they're not interface functions, why is this PR still moving them into `wesnoth.interface`?

Reply:
Huh? I thought I already reverted that bit… did I miss something?